srtcore/channel.cpp

```diff
@@ -522,20 +535,23 @@ EReadStatus CChannel::recvfrom(sockaddr* addr, CPacket& packet) const
     // value one Windows than 0, unless this procedure below is rewritten
     // to use WSARecvMsg().
 
-    DWORD size = CPacket::HDR_SIZE + packet.getLength();
+    int recv_ret = SOCKET_ERROR;
     DWORD flag = 0;
-    int addrsize = m_iSockAddrSize;
 
-    int msg_flags = 0;
-    int sockerror = ::WSARecvFrom(m_iSocket, (LPWSABUF)packet.m_PacketVector, 2, &size, &flag, addr, &addrsize, NULL, NULL);
-    int res;
-    if (sockerror == 0)
+    if (select_ret > 0)     // the total number of socket handles that are ready
     {
-        res = size;
+        DWORD size = (DWORD) (CPacket::HDR_SIZE + packet.getLength());
+        int addrsize = m_iSockAddrSize;
+
+        recv_ret = ::WSARecvFrom(m_iSocket, (LPWSABUF)packet.m_PacketVector, 2, &size, &flag, addr, &addrsize, NULL, NULL);
+        if (recv_ret == 0)
+            recv_size = size;
     }
-    else // == SOCKET_ERROR
+
+    // We do not handle <= SOCKET_ERROR as they are handled further by checking the recv_size
+    if (select_ret == SOCKET_ERROR || recv_ret == SOCKET_ERROR) // == SOCKET_ERROR
```

Comment:
`select_ret == -1`

Reply:
SOCKET_ERROR = -1 is defined in WinSock2.h, and is documented to be returned by [WSARecvFrom](https://docs.microsoft.com/en-us/windows/desktop/api/winsock2/nf-winsock2-wsarecvfrom) and [select](https://docs.microsoft.com/en-us/windows/desktop/api/winsock2/nf-winsock2-select) system functions on Windows.
I would suggest to use SOCKET_ERROR instead of -1.